### ISSUE TITLE
Failing to run ipex-llm ollama on Intel Arc A770

### ISSUE THREAD

**Issue by dan9070** [intel-analytics/ipex-llm#10995]
`2024/05/12 03:51:37 ERROR unable to locate llm runner directory.  Set OLLAMA_RUNNERS_DIR to the location of 'ollama_runners'`
This is an error I receive every thing upon running the ollama for llm-cpp environment. If this is not properly set and no other version of ollama is installed other than ipex-llm's, this error occurs:
`Error: [0] server cpu not listed in available servers map[]`

This was with a properly set-up environment, following every instruction in both setting up llama.cpp's symlinks and then ollama's. Everything was properly installed in llm-cpp in miniconda3, and the oneapi environment was activated.

**Comment by Jrizzos** [comment]
Hello, same here. Tried oneapi through pip and win installer but same results. Llama.cpp tested and runs just fine with SYCL support recognizing my Intel gpu. Ollama on the other hand does not recognize the gpu st all.My host is running win11.

**Comment by rnwang04** [comment]
Hi @dan9070  & @Jrizzos , 
we have reproduced this error on windows and we are trying to fix it. Once it is done, will update here to let you know.
For now, as a workaround, you can try `pip install ipex-llm[cpp]==2.1.0b20240508`, this version works on Windows.

**Comment by tardigrade34** [comment]
I'm having the same problem, looking forward to the Update.

**Comment by dan9070** [comment]
> Hi @dan9070 & @Jrizzos , we have reproduced this error (`Error: [0] server cpu not listed in available servers map[]`) on Windows and we are trying to fix it. Once it is done, will update here to let you know. For now, as a workaround, you can try `pip install ipex-llm[cpp]==2.1.0b20240508`, this version works on Windows.

This resolved it for me. Thank you.
Another person inside Intel Dev Inisders Discord also got it resolved with this package. I will notify other Intel Arc users with similar issues.

**Comment by rnwang04** [comment]
Hi all, this issue has been fixed for `ipex-llm[cpp]>=2.1.0b20240513` : )

**Comment by rnwang04** [comment]
> 2.1.0b20240513 doesn't work for me. Same error again: `Error: [0] server cpu not listed in available servers map`

Hi @tardigrade34 , I want to make sure if you have deleted the old `ollama.exe` in your directory and re-called `init-ollama.bat` with 2.1.0b20240513, there should be an `ollama.exe` and a `dist` directory in your current directory.

If you have already done this but still encounter problems, could you please provide us with the specific machine model and detailed error log so that we can further locate the problem ?

**Comment by czfdlut** [comment]
> Error: [0] server cpu not listed in available servers map

I compile the newest ollama，generate the binary：ollama.exe, app.exe。I start ollama.exe and app.exe，when  I run “ollama.exe run qwen:7b”，encounter the above error

**Comment by rnwang04** [comment]
> > Error: [0] server cpu not listed in available servers map
> I compile the newest ollama，generate the binary：ollama.exe, app.exe。I start ollama.exe and app.exe，when I run “ollama.exe run qwen:7b”，encounter the above error

Hi @czfdlut , ipex-llm ollma doesn't release `app.exe`, in my understanding, you compile ollama by yourself, right ?
Maybe you can try with ipex-llm ollama based on this quickstart(https://ipex-llm.readthedocs.io/en/latest/doc/LLM/Quickstart/ollama_quickstart.html).

**Comment by tardigrade34** [comment]
> > 2.1.0b20240513 doesn't work for me. Same error again: `Error: [0] server cpu not listed in available servers map`
> 
> Hi @tardigrade34 , I want to make sure if you have deleted the old `ollama.exe` in your directory and re-called `init-ollama.bat` with 2.1.0b20240513, there should be an `ollama.exe` and a `dist` directory in your current directory: ![image](https://private-user-images.githubusercontent.com/105281011/330354166-b2bcc057-6ae7-477b-926c-195e6027a9aa.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU2OTcxMjMsIm5iZiI6MTcxNTY5NjgyMywicGF0aCI6Ii8xMDUyODEwMTEvMzMwMzU0MTY2LWIyYmNjMDU3LTZhZTctNDc3Yi05MjZjLTE5NWU2MDI3YTlhYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUxNFQxNDI3MDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hMzQ2YjgwMWUzMTZiYjlkYzBkMzc3OThmMTE5YmZmZmFjZGVhZmYzZWVjZWZmZTI5MWU3ZjhhYTdjNTQwMTE1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Q0uApK_RiyCFZZPvEMxK_hP3I9sGVg0fyPjWWek3wd4)
> 
> If you have already done this but still encounter problems, could you please provide us with the specific machine model and detailed error log so that we can further locate the problem ?

I did what you said and it worked. Thanks a lot!

**Comment by rnwang04** [comment]
I will close this issue first, if you meet any other issues, you can reopen it or raise a new issue about it.
